### PR TITLE
feat(Zapping TV): add presence

### PR DESCRIPTION
**Pull request by sebalvaro**
## Description 

Add presence for Zapping TV.

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image

image

</details>

### CODE REVIEW

File: websites/Z/ZappingTV/presence.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+			).textContent.trim(),
+		};
+	},
+	assets: {
+		logo: string;
+		sleep: string;
+	} = {
+		logo: "https://i.imgur.com/5gyFfcd.png",
+		sleep: "https://i.imgur.com/gWBjy43.png",
+	};
```
Comment (theusaf) [inline websites/Z/ZappingTV/presence.ts:40]: Prefer an `enum` over this:
https://github.com/PreMiD/Presences/blob/63c0d808d092ca9e4a40feadd8caba231c4d2a5c/websites/C/Crunchyroll/presence.ts#L52-L58

File: websites/Z/ZappingTV/presence.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+		case "login": {
+			presenceData.details = "Iniciando sesión";
+			presence.setActivity(presenceData);
+			break;
+		}
+
+		case "": {
+			presenceData.details = "En la página principal";
+			presence.setActivity(presenceData);
```
Comment (theusaf) [inline websites/Z/ZappingTV/presence.ts:81]: Duplicated calls to `presence.setActivity` can be replaced by doing this at the end of the function instead:
https://github.com/PreMiD/Presences/blob/63c0d808d092ca9e4a40feadd8caba231c4d2a5c/websites/C/Crunchyroll/presence.ts#L214-L215

File: websites/Z/ZappingTV/presence.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+			smallImageText: "Zapping TV",
+			largeImageKey: zappingLogo,
+			startTimestamp: elapsed,
+		},
+		coloredLogos: boolean = await presence.getSetting("coloredLogos");
+
+	switch (document.location.pathname.split("/")[1]) {
+		case "player": {
+			const { logo, title, channel } = await getChannelData({ coloredLogos });
+			if (channel !== "" && title !== "" && logo !== "") {
```
Comment (theusaf) [inline websites/Z/ZappingTV/presence.ts:54]: Non-empty strings are truthy.
```suggestion
			if (channel && title && logo) {
```

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/Z/ZappingTV/presence.ts (hunk 1 above)
Comment (sebalvaro) [reply websites/Z/ZappingTV/presence.ts]: https://github.com/sebalvaro/Presences/blob/8bb7ebba3a0773052695650f4f08324ce64da0ce/websites/Z/ZappingTV/presence.ts#L34-L37

Comment (sebalvaro) [reply websites/Z/ZappingTV/presence.ts]: Done

File: websites/Z/ZappingTV/presence.ts (hunk 3 above)
Comment (sebalvaro) [reply websites/Z/ZappingTV/presence.ts]: Done :)

File: websites/Z/ZappingTV/presence.ts (hunk 2 above)
Comment (sebalvaro) [reply websites/Z/ZappingTV/presence.ts]: Done

File: websites/Z/ZappingTV/presence.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+				: logo.src,
+			title: getElement(".control-title").textContent.trim(),
+			channel: getElement(
+				"li.btn-channel.active > div.channel-data > div.name"
+			).textContent.trim(),
+		};
+	};
+enum Assets {
+	zappingLogo = "https://i.imgur.com/5gyFfcd.png",
+	zappingSleep = "https://i.imgur.com/gWBjy43.png",
```
Comment (DarkVillager) [inline websites/Z/ZappingTV/presence.ts:36]:
```suggestion
	ZappingLogo = "https://i.imgur.com/5gyFfcd.png",
	ZappingSleep = "https://i.imgur.com/gWBjy43.png",
```

File: websites/Z/ZappingTV/presence.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			smallImageText: "Zapping TV",
+			largeImageKey: Assets.zappingLogo,
```
Comment (DarkVillager) [inline websites/Z/ZappingTV/presence.ts:42]:
```suggestion
			largeImageKey: Assets.ZappingLogo,
```

File: websites/Z/ZappingTV/presence.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+				presenceData.largeImageKey = logo;
+				presenceData.smallImageKey = Assets.zappingLogo;
+				presenceData.endTimestamp = getEndTime();
+			}
+			break;
+		}
+
+		case "dashboard": {
+			presenceData.details = "En el Dashboard";
+			presenceData.smallImageKey = Assets.zappingSleep;
```
Comment (DarkVillager) [inline websites/Z/ZappingTV/presence.ts:63]:
```suggestion
			presenceData.smallImageKey = Assets.ZappingSleep;
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/Z/ZappingTV/presence.ts (hunk 4 above)
Comment (sebalvaro) [reply websites/Z/ZappingTV/presence.ts]: Ty & Done 👍🏼

File: websites/Z/ZappingTV/presence.ts (hunk 5 above)
Comment (sebalvaro) [reply websites/Z/ZappingTV/presence.ts]: Ty & Done 👍🏼

File: websites/Z/ZappingTV/presence.ts (hunk 6 above)
Comment (sebalvaro) [reply websites/Z/ZappingTV/presence.ts]: Ty & Done 👍🏼

**Comment by DarkVillager** [comment]
Remove changes to https://github.com/PreMiD/Presences/pull/7069/files#diff-053150b640a7ce75eff69d1a22cae7f0f94ad64ce9a855db544dda0929316519

**Comment by EncryptedDev** [comment]
Low quality. Think about supporting a wider range of pages

**Comment by sebalvaro** [comment]
> Low quality. Think about supporting a wider range of pages

It's basically a PWA. I think the presence doesn't need a wide range of pages since it only have a video player with dynamic content (like EPG, Channel Names and Hours).

I would probably add the homepage bc the user dashboard is included.

**Pull request closed without merging**